app/Helpers/database/player-achievement.php

```diff
@@ -485,22 +492,52 @@ function getAchievementDistribution(
     }
 
     // Returns an array of the number of players who have achieved each total, up to the max.
-    $query = "
+    if (env('FEATURE_AGGREGATE_QUERIES')) {
+        if ($flag === AchievementFlag::OfficialCore) {
+            $countColumn = $hardcore ? 'achievements_unlocked_hardcore' : 'achievements_unlocked';
+            $query = "SELECT pg.$countColumn AS AwardedCount, COUNT(*) AS NumUniquePlayers
+                      FROM player_games AS pg
+                      $joinStatementNew
+                      WHERE pg.game_id = :gameId AND pg.$countColumn > 0
+                      $requestedByStatement
+                      GROUP BY AwardedCount
+                      ORDER BY AwardedCount DESC";
+            unset($bindings['achievementFlag']);
+        } else {
+            $hardcoreStatement = $hardcore ? 'AND pa.unlocked_hardcore_at IS NOT NULL' : '';
+            $query = "SELECT InnerTable.AwardedCount AS AwardedCount, COUNT(*) AS NumUniquePlayers
+                    FROM (
+                        SELECT COUNT(*) AS AwardedCount
+                        FROM player_achievements AS pa
+                        LEFT JOIN Achievements AS ach ON ach.ID = pa.achievement_id
+                        $joinStatementNew
+                        WHERE ach.GameID = :gameId
+                        $hardcoreStatement
+                        AND ach.Flags = :achievementFlag
+                        $requestedByStatement
+                        GROUP BY pa.user_id
+                        ORDER BY AwardedCount DESC
+                    ) AS InnerTable
+                    GROUP BY InnerTable.AwardedCount";
```

Comment:
**game/1446**
With feature: 310ms
Without feature: 91ms

The new tables don't provide any aggregate data for unofficial achievements. Apparently, calculating that from the `player_achievements` table is slower than doing so from the `Awarded` table.

Reply:
Might be the missing index on `unlocked_hardcore_at`, i'll add that.